cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/spark/SparkTransformExecutorFactory.java

```diff
@@ -29,11 +30,17 @@
  * @param <T> the type of input for the created transform executors
  */
 public class SparkTransformExecutorFactory<T> extends TransformExecutorFactory<T> {
+  private final PluginContext pluginContext;
+  private final long logicalStartTime;
   private final Map<String, String> runtimeArgs;
 
-  public SparkTransformExecutorFactory(PluginContext pluginContext, Metrics metrics, long logicalStartTime,
+  public SparkTransformExecutorFactory(PluginContext pluginContext,
+                                       PipelinePluginInstantiator pluginInstantiator,
+                                       Metrics metrics, long logicalStartTime,
                                        Map<String, String> runtimeArgs) {
-    super(pluginContext, metrics, logicalStartTime);
+    super(pluginInstantiator, metrics);
+    this.pluginContext = pluginContext;
+    this.logicalStartTime = logicalStartTime;
```

Comment:
just a difference in how MapReduceRuntimeContext and SparkBatchRuntimeContext constructors are.  One takes an explicit logical start time, the other delegates to some other context